src/GeositeFramework/js/Map.js

```diff
@@ -216,22 +228,15 @@ require(['use!Geosite',
                 x.xmin, x.ymin, x.xmax, x.ymax,
                 new SpatialReference({ wkid: x.spatialReference.wkid })
             );
-        view.esriMap.setExtent(extent);
+        view.esriMapView.extent = extent;
     }
 
     function saveExtent(view) {
-        view.model.set('extent', view.esriMap.extent);
+        view.model.set('extent', view.esriMapView.extent);
     }
 
     function selectBasemap(view) {
-        // Hide the current basemap layer
-        if (view.currentBasemapLayer !== undefined) {
-            view.currentBasemapLayer.hide();
-        }
-        // Show the new basemap layer (at index 0)
-        view.currentBasemapLayer = view.model.getSelectedBasemapLayer(view.esriMap);
-        view.currentBasemapLayer.show();
-        view.esriMap.reorderLayer(view.currentBasemapLayer, 0);
+        view.model.getSelectedBasemapLayer(view.esriMap);
```

Comment:
This change is a little opaque as well.  It looks like it's taking advantage of some side effects in `getSelectedBasemapLayer` which is kind of confusing.  Was the rest of the code superfluous?  Consider renaming the method if it is actually more like `setBasemapLayer` or something more accurate.